stats/opentelemetry/e2e_test.go

```diff
@@ -582,3 +612,348 @@ func pollForWantMetrics(ctx context.Context, t *testing.T, reader *metric.Manual
+
+	// Make two RPC's, a unary RPC and a streaming RPC. These should cause
+	// certain metrics and traces to be emitted.
+	if _, err := ss.Client.UnaryCall(ctx, &testpb.SimpleRequest{Payload: &testpb.Payload{
+		Body: make([]byte, 10000),
+	}}, grpc.UseCompressor(gzip.Name)); err != nil { // Deterministic compression.
+		t.Fatalf("Unexpected error from UnaryCall: %v", err)
+	}
+	stream, err := ss.Client.FullDuplexCall(ctx)
+	if err != nil {
+		t.Fatalf("ss.Client.FullDuplexCall failed: %f", err)
+	}
+
+	stream.CloseSend()
+	if _, err = stream.Recv(); err != io.EOF {
+		t.Fatalf("stream.Recv received an unexpected error: %v, expected an EOF error", err)
+	}
+
+	// Verify metrics
+	rm := &metricdata.ResourceMetrics{}
+	reader.Collect(ctx, rm)
+
+	gotMetrics := map[string]metricdata.Metrics{}
+	for _, sm := range rm.ScopeMetrics {
+		for _, m := range sm.Metrics {
+			gotMetrics[m.Name] = m
+		}
+	}
+
+	wantMetrics := testutils.MetricData(testutils.MetricDataOptions{
+		Target:                     ss.Target,
+		UnaryCompressedMessageSize: float64(57),
+	})
+	testutils.CompareMetrics(ctx, t, reader, gotMetrics, wantMetrics)
+
+	// Verify traces
+	spans := exporter.GetSpans()
+	if got, want := len(spans), 6; got != want {
+		t.Fatalf("Got %d spans, want %d", got, want)
+	}
+
+	// Add assertions for specific span attributes and events as needed.
+	// For example, to check if the client span has the correct status:
+	clientSpan := spans[2]
+	if got, want := clientSpan.Status.Code, otelcodes.Ok; got != want {
+		t.Errorf("Got status code %v, want %v", got, want)
+	}
+}
+
+// TestSpan verifies that the gRPC Trace Binary propagator
+// correctly propagates span context between a client and server using the
+// grpc-trace-bin header. It sets up a stub server with OpenTelemetry tracing
+// enabled, makes a unary RPC.
+func (s) TestSpan(t *testing.T) {
+	// Using defaultTraceOptions to set up OpenTelemetry with an in-memory exporter
+	traceOptions, spanExporter := defaultTraceOptions(t)
+	testSpan(t, traceOptions, spanExporter)
+}
+
+// testSpan verifies that the TextMapPropagator correctly propagates span context
+// between a client and server using the grpc-trace-bin header. It sets up a stub
+// server with OpenTelemetry tracing enabled, performs a unary RPC and a streaming
+// RPC, and checks that the expected number of spans are created with the correct
+// attributes and events. The function also validates that the trace IDs and span
+// IDs are consistent between client and server spans.
+func testSpan(t *testing.T, traceOptions *TraceOptions, spanExporter *tracetest.InMemoryExporter) {
+	// Start the server with OpenTelemetry options
+	ss := setupStubServer(t, nil, traceOptions)
+	defer ss.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	// Create a parent span for the client call
+	ctx, _ = otel.Tracer("grpc-open-telemetry").Start(ctx, "test-parent-span")
```

Comment:
same thing, why do we have to inject a parent span outside. This test should exactly look how a user would configure the plugin to instrument the code and we should be able to verify if required values are being emitted.